### PR TITLE
javascript-javascript3-week2

### PR DESCRIPTION
**Pull request by suvarna-ratna**
feedback?

### CODE REVIEW

File: javascript/javascript3/week2/promise.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+//Promise that resolves after set time:
+function promiseResolveAfterSetTime(resolveAfter) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
```
Comment (flom84) [inline javascript/javascript3/week2/promise.js:5]: There is a small mistake. A string argument is need for `resolve` function.
`Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?ts(2794)`
For example `resolve('timeout after x seconds')`

File: javascript/javascript3/week2/promise.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    console.log(error);
+  }
+}
+usingAsyncAwait();
+
+//Rewrite time:
+function setTimeoutPromise(resolveAfetr) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
```
Comment (flom84) [inline javascript/javascript3/week2/promise.js:29]: There is a small mistake. A string argument is need for resolve function.
Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?ts(2794)

File: javascript/javascript3/week2/promise.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    console.log(error);
+  });
+
+//Fetching and waiting:
+//Wait 3 seconds
+//After 3 seconds fetch some data from any api you like
+//Log out the data from the api
+const promise = new Promise((resolve) => {
+  setTimeout(() => {
+    resolve();
```
Comment (flom84) [inline javascript/javascript3/week2/promise.js:66]: There is a small mistake. A string argument is need for resolve function.
`Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?ts(2794)`
For example resolve('timeout after x seconds')

**Review by flom84 — CHANGES_REQUESTED** [review]
- Visual promise program doesn't seem to work. The balls don't move to their final destination. For development purposes and debugging you can use https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer
image

File: javascript/javascript3/week2/promise.js (hunk 3 above)
Comment (suvarna-ratna) [reply javascript/javascript3/week2/promise.js:66]: done

**Pull request merged**